src/classes/BDI_DataImportService.cls

```diff
@@ -81,7 +81,11 @@ global with sharing class BDI_DataImportService {
 
         List<DataImportBatch__c> dataImportBatches = BDI_DataImportBatch_SEL.selectByIds(new Set<Id>{ batchId });
         if (dataImportBatches.size() == 1) {
-            return dataImportBatches[0];
+            DataImportBatch__c batch = dataImportBatches[0];
+            if (batch.Active_Fields__c != null) {
+                relevantDataImportFields = BGE_BatchGiftEntry_UTIL.getDataImportFields(batchId, false);
```

Comment:
I'm not comfortable with creating a dependency on BGE_BatchGiftEntryUtil here. Should we get the selected fields as part of diSettingsFromDiBatch(DataImportBatch__c dataImportBatch) instead?

Reply:
That method returns a Data_Import_Settings__c object, which doesn't have Active Fields on it. Is there another way we could get around this? Should the dependency go the other way? BGE is the only place we use the Active Fields JSON, so parsing it in the BGE UTIL seemed logical.

Reply:
I can live with it.